<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#-4장-2절-모수-추정과-신뢰구간-(Parameter-Estimation-&amp;-Confidence-Interval)" data-toc-modified-id="-4장-2절-모수-추정과-신뢰구간-(Parameter-Estimation-&amp;-Confidence-Interval)-1"><span class="toc-item-num">1&nbsp;&nbsp;</span> 4장 2절 모수 추정과 신뢰구간 (Parameter Estimation &amp; Confidence Interval)</a></span></li><li><span><a href="#4장-2절-모수-추정과-신뢰구간" data-toc-modified-id="4장-2절-모수-추정과-신뢰구간-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>4장 2절 모수 추정과 신뢰구간</a></span><ul class="toc-item"><li><span><a href="#(Parameter-Estimation-&amp;-Confidence-Interval)" data-toc-modified-id="(Parameter-Estimation-&amp;-Confidence-Interval)-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>(Parameter Estimation &amp; Confidence Interval)</a></span><ul class="toc-item"><li><span><a href="#📝-[문제-상황]" data-toc-modified-id="📝-[문제-상황]-2.1.1"><span class="toc-item-num">2.1.1&nbsp;&nbsp;</span>📝 [문제 상황]</a></span></li><li><span><a href="#🔍-소문항-1:-점추정치(Point-Estimate)-구하기" data-toc-modified-id="🔍-소문항-1:-점추정치(Point-Estimate)-구하기-2.1.2"><span class="toc-item-num">2.1.2&nbsp;&nbsp;</span>🔍 소문항 1: 점추정치(Point Estimate) 구하기</a></span></li><li><span><a href="#📊-소문항-2:-95%-신뢰구간-산출-및-해석" data-toc-modified-id="📊-소문항-2:-95%-신뢰구간-산출-및-해석-2.1.3"><span class="toc-item-num">2.1.3&nbsp;&nbsp;</span>📊 소문항 2: 95% 신뢰구간 산출 및 해석</a></span><ul class="toc-item"><li><span><a href="#1.-계산-과정" data-toc-modified-id="1.-계산-과정-2.1.3.1"><span class="toc-item-num">2.1.3.1&nbsp;&nbsp;</span>1. 계산 과정</a></span></li></ul></li></ul></li></ul></li></ul></div>

<H1> 4장 2절 모수 추정과 신뢰구간 (Parameter Estimation & Confidence Interval)

# 4장 2절 모수 추정과 신뢰구간
## (Parameter Estimation & Confidence Interval)

---

### 📝 [문제 상황]
**A 전자 기기 생산 라인**에서 생산된 제품의 품질을 점검하기 위해 100개의 샘플을 무작위로 추출하였습니다. 조사 결과, 100개의 제품 중 8개가 불량품으로 판정되었습니다. (**유의수준 $\alpha = 0.05$**)

* **표본 크기 ($n$):** 100
* **불량품 수 ($x$):** 8

---

### 🔍 소문항 1: 점추정치(Point Estimate) 구하기

점추정치는 표본 데이터를 바탕으로 모집단의 모수를 하나의 수치로 추정하는 것입니다. 여기서는 표본 불량률($\hat{p}$)이 모집단 불량률($p$)의 점추정치가 됩니다.

$$\hat{p} = \frac{x}{n} = \frac{8}{100} = 0.08$$

> **결과:** 이 생산 라인 전체 불량률($p$)에 대한 점추정치는 **0.08 (또는 8%)**입니다.

---

### 📊 소문항 2: 95% 신뢰구간 산출 및 해석

[Image of Normal distribution showing 95% confidence interval and critical values Z=1.96]

모비율에 대한 신뢰구간은 표본의 크기가 충분히 크므로 ($np \ge 5, n(1-p) \ge 5$) 정규분포 근사를 사용하여 산출합니다. 95% 신뢰수준에서의 임계값 $z_{0.025}$는 약 **1.96**입니다.

#### 1. 계산 과정
신뢰구간 공식:
$$\hat{p} \pm z_{\alpha/2} \sqrt{\frac{\hat{p}(1-\hat{p})}{n}}$$

* **오차 한계 (Margin of Error):**
    $$1.96 \times \sqrt{\frac{0.08 \times 0.92}{100}} \approx 1.96 \times 0.0271 \approx 0.0531$$

* **신뢰구간 산출:**
    * 하한값: $0.08 - 0.0531 = 0.0269$
    * 상한값: $0.08 + 0.0531 = 0.1331$


In [2]:
import pandas as pd
import statsmodels.stats.api as sms

# 1. 데이터 로드
df = pd.read_csv('https://raw.githubusercontent.com/ADPclass/ADP_book_ver01/main/data/delivery_data.csv')
a_time = df[df['company'] == 'A']['delivery_time']

# 2. 점추정 (pandas 활용)
point_mean = a_time.mean()
print(f"점추정치(평균): {point_mean:.2f}분")

# 3. 구간추정 (statsmodels의 DescrStatsW 활용)
# 데이터를 DescrStatsW에 담고, .tconfint_mean()만 호출하면 끝!
res = sms.DescrStatsW(a_time)
conf_int_mean = res.tconfint_mean(alpha=0.05)

print(f"95% 신뢰구간: {conf_int_mean}")

점추정치(평균): 134.70분
95% 신뢰구간: (131.51255388030737, 137.8971592156926)


In [3]:

import pandas as pd
import statsmodels.stats.api as sms

# 데이터 로드
df = pd.read_csv('https://raw.githubusercontent.com/ADPclass/ADP_book_ver01/main/data/manufacturing_quality.csv')

# 1. 빈도 파악
n_total = 100
n_fail = 8

# 2. 점추정
p_hat = n_fail / n_total
print(f"불량률 점추정치: {p_hat:.4f}")

# 3. 구간추정
# alpha=0.05는 95% 신뢰구간을 의미합니다.
conf_int_prop = sms.proportion_confint(count=n_fail, 
                                       nobs=n_total, 
                                       alpha=0.05, 
                                       method='normal')

print(f"불량률 95% 신뢰구간: {conf_int_prop}")

불량률 점추정치: 0.0800
불량률 95% 신뢰구간: (0.026827510007233275, 0.13317248999276673)


[소문항 1 답안]표본 100개 중 불량품이 8개이므로 표본 비율 p = 8/100 = 0.08이다. \
표본 비율은 모비율의 불편추정량이므로, 해당 생산 라인 전체의 불량률에 대한 **점추정치는 0.08(또는 8%)**이다.

[소문항 2 답안]95% 신뢰수준 하에서 전체 불량률에 대한 신뢰구간을 산출한 결과 [0.0268, 0.1332]이다. \
이는 모집단인 생산 라인 전체의 실제 불량률이 약 2.68%에서 13.32% 사이에 존재할 것이라고 95% 신뢰할 수 있음을 의미한다.